### PR TITLE
DAC-71 

### PR DESCRIPTION
**Pull request by TimothyNecr0**
Add the function of the JWT verification to the button

### CODE REVIEW

File: DD/Views/Partials/homePage/_redirectButton.cshtml [hunk 1]
```diff
@@ -1,16 +1,65 @@
...
     </a>
 }
+
+<script>
+    document.getElementById('homeRedirectButton').addEventListener('click', function () {
+        const token = localStorage.getItem('authToken'); // Get the token from localStorage
+
+        if (!token) {
+            alert('No tienes una sesion activa. Por favor, inicia sesion o registrate.');
+            window.location.href = '/loginpage'; // Redirect to login page if there is no token
```
Comment (rgr182) [inline DD/Views/Partials/homePage/_redirectButton.cshtml:29]: get value from viewbag or at least use constants

File: DD/Views/Partials/homePage/_redirectButton.cshtml [hunk 2]
```diff
@@ -1,16 +1,65 @@
...
+            } else {
+                throw new Error('Error en la validacion del token');
+            }
+        })
+        .then(data => {
+            console.log('Response data:', data); // Check backend response
+            if (data.Message) {
+                alert(data.Message); // Show message with remaining token minutes
+            } else {
+                alert('Se encontro una sesion activa'); // Default message if no message is returned
```
Comment (rgr182) [inline DD/Views/Partials/homePage/_redirectButton.cshtml:55]: Constants

File: DD/Views/Partials/homePage/_redirectButton.cshtml [hunk 3]
```diff
@@ -1,16 +1,65 @@
...
+            }
+            window.location.href = 'https://chatbot.bolsadetrabajo.com'; // this need to be changed to the real URL
+
+        .catch(error => {
+            console.error('Error while validating the token:', error);
+            alert('Tu sesion ha expirado o no estas autenticado. Por favor, inicia sesion o registrate.');
+            window.location.href = '/loginpage'; // Redirect to login page
```
Comment (rgr182) [inline DD/Views/Partials/homePage/_redirectButton.cshtml:62]: Constants

**Review by rgr182 — CHANGES_REQUESTED** [review]

File: DD/Views/Partials/homePage/_redirectButton.cshtml [hunk 4]
```diff
@@ -1,16 +1,88 @@
...
+            }
+        })
+        .then(data => {
+            console.log('Response data:', data); // Check backend response
+            if (data.Message) {
+                alert(data.Message); // Show message with remaining token minutes
+            } else {
+                alert('Se encontro una sesion activa'); // Default message if no message is returned
+            }
+            window.location.href = 'https://chatbot.bolsadetrabajo.com'; // Redirect to chatbot if the token is valid, change this later
```
Comment (Paroxyse) [inline DD/Views/Partials/homePage/_redirectButton.cshtml:80]: Use @Model.HomeRedirectButtonUrl.Url for this url

**Pull request merged**